drivers/video/arducam_mega.c

```diff
@@ -0,0 +1,1064 @@
+	struct k_work_sync work_sync;
+	struct k_poll_signal *signal;
+	struct arducam_mega_info *info;
+	struct mega_sdk_data ver;
+	uint8_t fifo_first_read;
+	uint32_t fifo_length;
+	uint8_t stream_on;
+};
+
+static struct arducam_mega_info mega_infos[] = {{
+							.support_resoultion = 7894,
+							.support_special_effects = 63,
+							.exposure_value_max = 30000,
+							.exposure_value_min = 1,
+							.gain_value_max = 1023,
+							.gain_value_min = 1,
+							.enable_focus = 1,
+							.enable_sharpness = 0,
+							.device_address = 0x78,
+						},
+						{
+							.support_resoultion = 7638,
+							.support_special_effects = 319,
+							.exposure_value_max = 30000,
+							.exposure_value_min = 1,
+							.gain_value_max = 1023,
+							.gain_value_min = 1,
+							.enable_focus = 0,
+							.enable_sharpness = 1,
+							.device_address = 0x78,
+						}};
+
+#define ARDUCAM_MEGA_VIDEO_FORMAT_CAP(width, height, format)                                       \
+	{                                                                                          \
+		.pixelformat = (format), .width_min = (width), .width_max = (width),               \
+		.height_min = (height), .height_max = (height), .width_step = 0, .height_step = 0  \
+	}
+
+static struct video_format_cap fmts[] = {
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(96, 96, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(128, 128, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(320, 240, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(320, 320, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(640, 480, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1280, 720, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1600, 1200, VIDEO_PIX_FMT_RGB565),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1920, 1080, VIDEO_PIX_FMT_RGB565),
+	{0},
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(96, 96, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(128, 128, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(320, 240, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(320, 320, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(640, 480, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1280, 720, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1600, 1200, VIDEO_PIX_FMT_JPEG),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1920, 1080, VIDEO_PIX_FMT_JPEG),
+	{0},
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(96, 96, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(128, 128, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(320, 240, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(320, 320, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(640, 480, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1280, 720, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1600, 1200, VIDEO_PIX_FMT_YUYV),
+	ARDUCAM_MEGA_VIDEO_FORMAT_CAP(1920, 1080, VIDEO_PIX_FMT_YUYV),
+	{0},
+	{0},
+};
+
+#define SUPPORT_RESOULTION_NUM 9
+
+static uint8_t support_resoultion[SUPPORT_RESOULTION_NUM] = {
```

Comment:
Make const

Reply:
@bjarki-trackunit 
Thank you for your detail check. This is the same as the [fmts](https://github.com/zephyrproject-rtos/zephyr/pull/66994#discussion_r1436182143)， due to the arducam mega driver supports multi cameras.

Reply:
ok